agent/api/containerstatus.go

```diff
@@ -151,6 +180,38 @@ func (cs *ContainerStatus) MarshalJSON() ([]byte, error) {
 	return []byte(`"` + cs.String() + `"`), nil
 }
 
+// UnmarshalJSON overrides the logic for parsing the JSON-encoded container health data
+func (healthStatus *ContainerHealthStatus) UnmarshalJSON(b []byte) error {
+	*healthStatus = ContainerHealthUnknown
+
+	if strings.ToLower(string(b)) == "null" {
+		return nil
+	}
+	if b[0] != '"' || b[len(b)-1] != '"' {
+		return errors.New("container health status unmarshal: status must be a string or null; Got " + string(b))
+	}
+
+	strStatus := string(b[1 : len(b)-1])
+	switch strStatus {
+	case "UNKNOWN":
```

Comment:
nit: please add a comment here stating that the status is already set to `ContainerHealthUnknown`.